backend/tests/auth/test_device_router.py

```
@@ -0,0 +1,56 @@
+from base64 import b64encode
+from time import sleep
+
+ENDPOINT = "/api/auth/device"
+
+
+class TestAuthDeviceRouter:
+    @staticmethod
+    def basic_auth(username, password):
+        token = b64encode(f"{username}:{password}".encode("utf-8")).decode("ascii")
+        return f"Basic {token}"
+
+    def test_get_device_token(self, client):
+        response = client.post(f"{ENDPOINT}/device_token?client_id=test")
+        assert response.status_code == 200
+        assert response.json()["oidc_redirect_uri"] == "http://test-redirect-uri"
+        assert response.json()["status"] == "authorization_pending"
+
+    def test_get_jwt_token(self, client):
+        response = client.post(f"{ENDPOINT}/device_token?client_id=test")
+        device_code = response.json()["device_code"]
+        sleep(5)
```

Why is a sleep needed in the test? This will just slow down the test, no?

---

There is a built-in SlowdownException in the device flow logic, to stop people from non-stop polling. Without the sleep, the instant check throws this exception instead of executing actual logic. But yes, this does slow everything down :(